src/gui/pointcloud/qgspointcloudrendererpropertieswidget.cpp

```diff
@@ -124,13 +125,45 @@ QgsPointCloudRendererPropertiesWidget::QgsPointCloudRendererPropertiesWidget( Qg
   connect( mHorizontalTriangleThresholdSpinBox, qOverload<double>( &QgsDoubleSpinBox::valueChanged ), this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
   connect( mHorizontalTriangleUnitWidget, &QgsUnitSelectionWidget::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
 
-  // show label options only for virtual point clouds
-  bool showLabelOptions = !mLayer->dataProvider()->subIndexes().isEmpty();
-  mLabels->setVisible( showLabelOptions );
-  mLabelOptions->setVisible( showLabelOptions );
-  mLabelOptions->setDialogTitle( tr( "Customize label text" ) );
-  connect( mLabels, &QCheckBox::stateChanged, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
-  connect( mLabelOptions, &QgsFontButton::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+  // show virtual point cloud options only when vpc layer is selected
+  if ( !mLayer->dataProvider()->subIndexes().isEmpty() )
+  {
+    mLabelOptions->setDialogTitle( tr( "Customize label text" ) );
+    connect( mLabels, &QCheckBox::stateChanged, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+    connect( mLabelOptions, &QgsFontButton::changed, this, &QgsPointCloudRendererPropertiesWidget::emitWidgetChanged );
+    mZoomOutOptions->addItem( tr( "Show extents only" ), static_cast<int>( Qgis::PointCloudZoomOutRenderBehavior::RenderExtents ) );
+
+    if ( const QgsVirtualPointCloudProvider *vpcProvider = dynamic_cast<QgsVirtualPointCloudProvider *>( mLayer->dataProvider() ) )
+    {
+      if ( vpcProvider->overview() )
+      {
+        mZoomOutOptions->addItem( tr( "Show overview only" ), static_cast<int>( Qgis::PointCloudZoomOutRenderBehavior::RenderOverview ) );
+        mZoomOutOptions->addItem( tr( "Show extents over overview" ), static_cast<int>( Qgis::PointCloudZoomOutRenderBehavior::RenderOverviewAndExtents ) );
+      }
+    }
+    else
+    {
+      mZoomOutOptions->setEnabled( false );
+    }
+
+    connect( mZoomOutOptions, qOverload<int>( &QComboBox::currentIndexChanged ), this, [this]( int currentZoomOutOption ) {
+      switch ( static_cast<Qgis::PointCloudZoomOutRenderBehavior>( currentZoomOutOption ) )
```

Comment:
```suggestion
      switch ( mZoomOutOptions->currentData().value< Qgis::PointCloudZoomOutRenderBehavior >() )
```

Reply:
Can you tell me why this and not that? Wouldn't it be better to use the value passed into the lambda from signal?

Reply:
The original code assumes that the combo box index is identical to the integer value of the enum keys. That won't be the case if the combo box entries are re-ordered in future, so the original code is fragile. The new version uses the internal data attached to each individual entry, so won't need any changes if the entries are re-ordered.